// @nutui/nutui/dist/packages/_es/Collapse.js
import { ref, computed, watch, provide, openBlock, createElementBlock, normalizeClass, renderSlot } from "vue";
import { c as createComponent } from "./component-0fbad28e.js";
import { _ as _export_sfc } from "./_plugin-vue_export-helper-cc2b3d55.js";
import "../locale/lang";
const { create, componentName } = createComponent("collapse");
const _sfc_main = create({
  props: {
    modelValue: {
      type: [String, Number, Array],
      default: ""
    },
    accordion: {
      type: Boolean,
      default: false
    }
  },
  emits: ["update:modelValue", "change"],
  setup(props, { emit }) {
    const collapseDom = ref(null);
    const innerValue = ref(props.modelValue || (props.accordion ? "" : []));
    const classes = computed(() => {
      const prefixCls = componentName;
      return {
        [prefixCls]: true
      };
    });
    watch(
      () => props.modelValue,
      (val) => {
        innerValue.value = val;
      }
    );
    const changeVal = (val, name, status = true) => {
      innerValue.value = val;
      emit("update:modelValue", val);
      emit("change", val, name, status);
    };
    const updateVal = (name) => {
      if (props.accordion) {
        if (innerValue.value === name) {
          changeVal("", name, false);
        } else {
          changeVal(name, name, true);
        }
      } else {
        if (Array.isArray(innerValue.value)) {
          if (innerValue.value.includes(name)) {
            const newValue = innerValue.value.filter((v) => v !== name);
            changeVal(newValue, name, false);
          } else {
            const newValue = innerValue.value.concat([name]);
            changeVal(newValue, name, true);
          }
        } else {
          console.warn("[NutUI] <Collapse> 未开启手风琴模式时 v-model 应为数组");
        }
      }
    };
    const isExpanded = (name) => {
      if (props.accordion) {
        return innerValue.value === name;
      } else if (Array.isArray(innerValue.value)) {
        return innerValue.value.includes(name);
      }
      return false;
    };
    provide("collapseParent", {
      updateVal,
      isExpanded
    });
    return { collapseDom, classes };
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("view", {
    class: normalizeClass(_ctx.classes),
    ref: "collapseDom"
  }, [
    renderSlot(_ctx.$slots, "default")
  ], 2);
}
const index = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  index as default
};
