// @nutui/nutui/dist/packages/_es/SwipeGroup.js
import { ref, provide, openBlock, createElementBlock, renderSlot } from "vue";
import { c as createComponent } from "./component-0fbad28e.js";
import { _ as _export_sfc } from "./_plugin-vue_export-helper-cc2b3d55.js";
import "../locale/lang";
const { create } = createComponent("swipe-group");
const _sfc_main = create({
  props: {
    lock: {
      type: Boolean,
      default: false
    }
  },
  setup(props) {
    const swipeGroupRef = ref(null);
    const name = ref("");
    const update = (n) => {
      name.value = n;
    };
    provide("swipeGroup", {
      update,
      lock: props.lock,
      name
    });
    return { swipeGroupRef };
  }
});
const _hoisted_1 = {
  class: "nut-swipe-group",
  ref: "swipeGroupRef"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [
    renderSlot(_ctx.$slots, "default")
  ], 512);
}
const index = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  index as default
};
